cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -690,6 +700,244 @@ def refresh(self):
             self._gce_image['status'] = "unknown"
 
 
+class GCEInstance(BaseInstance):
+    # https://cloud.google.com/compute/docs/reference/latest/instances
+    # The status of the instance. One of the following values:
+    # PROVISIONING, STAGING, RUNNING, STOPPING, SUSPENDING, SUSPENDED,
+    # and TERMINATED.
+    INSTANCE_STATE_MAP = {
+        'PROVISIONING': InstanceState.PENDING,
+        'STAGING': InstanceState.PENDING,
+        'RUNNING': InstanceState.RUNNING,
+        'STOPPING': InstanceState.CONFIGURING,
+        'TERMINATED': InstanceState.TERMINATED,
+        'SUSPENDING': InstanceState.CONFIGURING,
+        'SUSPENDED': InstanceState.STOPPED
+    }
+
+    def __init__(self, provider, gce_instance):
+        super(GCEInstance, self).__init__(provider)
+        self._gce_instance = gce_instance
+
+    @property
+    def resource_url(self):
+        return self._gce_instance.get('selfLink')
+
+    @property
+    def id(self):
+        """
+        Get the instance identifier.
+        """
+        return self._gce_instance['id']
+
+    @property
+    def name(self):
+        """
+        Get the instance name.
+        """
+        return self._gce_instance['name']
+
+    @name.setter
+    # pylint:disable=arguments-differ
+    def name(self, value):
```

Comment:
@baizhang @mbookman @chiniforooshan @afgane I'm wondering whether we should consider silently ignoring this? On the one hand, setting the name does not appear to be a critical function to me , and if that rather innocuous action requires a try/exception block, that would seem to complicate the client interface and/or make it difficult to write cross-cloud code? On the other hand, there genuinely is a difference in functionality, and attempting to gloss over it may lull the user into a false sense of having succeeded. Perhaps the compromise is to document, at the interface level, that setting the name is not guaranteed to succeed, and code should not rely on the name being set?